backend/safe_condition_parser.py

```
@@ -389,12 +389,9 @@ def safe_evaluate_condition(
         return _parser.evaluate(expression, context)
-    except ValueError:
-        # Validation and safety errors from the parser should surface to callers/tests
-        raise
     except Exception as e:
-        # Unexpected errors -> log and re-raise as ValueError to keep the API consistent
+        # Log error and return safe default (False)
         import logging
         logger = logging.getLogger(__name__)
-        logger.error(f"Unexpected error evaluating condition '{expression}': {e}", exc_info=True)
-        raise ValueError("Unexpected error while evaluating condition") from e
+        logger.error(f"Error evaluating condition '{expression}': {e}")
```

Ruff (0.14.10)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)